executor/join.go

```diff
@@ -81,42 +94,32 @@ type probeWorker struct {
 	// for every naaj probe worker,  pre-allocate the int slice for store the join column index to check.
 	needCheckBuildRowPos []int
 	needCheckProbeRowPos []int
+	probeChkResourceCh   chan *probeChkResource
+	joinChkResourceCh    chan *chunk.Chunk
+	probeResultCh        chan *chunk.Chunk
 }
 
 // HashJoinExec implements the hash join algorithm.
 type HashJoinExec struct {
 	baseExecutor
+	*hashJoinCtx
 
 	probeSideTupleFetcher *probeSideTupleFetcher
-	*hashJoinCtx
-	probeWorkers      []probeWorker
+	probeWorkers          []probeWorker
+
 	buildSideExec     Executor
 	buildSideEstCount float64
-	outerFilter       expression.CNFExprs
 	probeKeys         []*expression.Column
 	probeNAKeys       []*expression.Column
 	buildKeys         []*expression.Column
-	buildNAKeys       []*expression.Column
```

Comment:
Why only move `buildNAKeys` into hashJoinCtx?

Reply:
I think the only usage below https://github.com/pingcap/tidb/pull/39155/files#diff-ac6f3fc5c91a1d77229d11140f9c8a8b561eb33ce595fc2d0ffd5d912cf8e3abR893 can use a bool to substitute